File: 100-million-calc/Ruby/HundredMillionCalc.rb
require 'benchmark'

# 計算回数
ITERATIONS = 100_000_000  # 1億回の繰り返し

puts "計算回数: #{ITERATIONS.to_s.gsub(/(\d)(?=(\d\d\d)+(?!\d))/, '\\1,')}"

result = 0.0
duration = Benchmark.measure do
  ITERATIONS.times do |i|
    result += Math.sin(i) * Math.cos(i)
  end
end


puts "計算結果: #{result}"

File: 100-million-calc/Python3/HundredMillionCalc.py
import math

def main():
    # 計算回数
    ITERATIONS = 100000000  # 1億回の繰り返し

    print(f"計算回数: {ITERATIONS:,}")
    
    # 時間のかかる処理
    result = 0.0
    for i in range(ITERATIONS):
        result += math.sin(i) * math.cos(i)
    
    print(f"計算結果: {result}")

if __name__ == "__main__":
    main()